addons/website_event/models/event.py

```diff
@@ -156,3 +156,10 @@ def _default_website_meta(self):
         res['default_twitter']['twitter:card'] = 'summary'
         res['default_meta_description'] = self.date_begin
         return res
+
+    @api.model
+    def update_events_state(self):
+        current_company = self.env.company_id
+        now = fields.Datetime.now()
+        for event in self.search([('company_id', '=', current_company.id), ('state', '!=', 'done'), ('date_begin', '<=', now)]):
+            event.state = 'done'
```

Comment:
You are writing on each event individually, why not on all events you found directly ?
```python
events = self.search(...)
events.write({'state': 'done'})
```

Also, is there a reason to do this only for the current company ? I suppose JKE/RDE told you to do it ?

Reply:
> You are writing on each event individually, why not on all events you found directly ?

Yes, indeed.

> Also, is there a reason to do this only for the current company ? I suppose JKE/RDE told you to do it ?

Well, in fact as I saw the explanation from YTI about the new company system, I copied what he did to be able to work with the right company we are on. No ?

Reply:
Mmh. Don't know, @JKE-be will check anyway. Adapt the events.write stuff, we will see afterwards.

Reply:
> You are writing on each event individually, why not on all events you found directly ?
> events = self.search(...)
> events.write({'state': 'done'})

done